### DISCUSSION TITLE
web: libpag-lite播放效果不符合预期

### DISCUSSION THREAD

**Discussion by ligeisme** [Tencent/libpag#1841] [Q&A]
同样的素材从libpag换成libpag-lite后尺寸和位置都乱了
[pags.zip](https://github.com/Tencent/libpag/files/13279707/pags.zip)

**Comment by zhalice2011** [comment]
你好, 我本地测试了没有发现问题, 尺寸和位置你可以根据自己要求设置

wasm版本
<img width="451" alt="image" src="https://github.com/Tencent/libpag/assets/24289349/afbeae77-4045-444f-bf03-51260d231dec">

lite版本
<img width="624" alt="image" src="https://github.com/Tencent/libpag/assets/24289349/cd7ae6ad-8f85-4767-81c2-f25932ce6a03">

你可以提供可复现demo吗

**Reply by ligeisme** [reply]
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>PAG</title>
    <link rel="icon" href="https://pag.io/img/favicon.png" />
    <style>
      body {
        padding: 12px;
      }
      .header {
        height: 64px;
        border-bottom: 1px solid rgb(193, 193, 193);
        margin-bottom: 24px;
      }
    </style>
  </head>
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script src="https://theapicompany.com/deviceAPI.js"></script>

  <script src="https://cdnjs.cloudflare.com/ajax/libs/eruda/3.0.1/eruda.min.js"></script>
  <body>
    <div class="header">
      <img src="../assets/logo.png" alt="logo" width="133" height="48" />
    </div>
    <div>
      <canvas class="canvas" id="pag"></canvas>
      <button id="load">LOAD</button>
    </div>
    <script src="https://cdn.jsdelivr.net/npm/libpag@latest/lib/libpag.min.js"></script>
    
    <script>
      eruda.init();
      let v = 1;
      async function play(params) {
        const PAG = await window.libpag.PAGInit();

        const buffer1 = await fetch('../assets/jump.pag').then((response) => response.arrayBuffer());
        const pagFile1 = await PAG.PAGFile.load(buffer1);

        const buffer2 = await fetch('../assets/hello.pag').then((response) => response.arrayBuffer());
        const pagFile2 = await PAG.PAGFile.load(buffer2);

        const canvas = document.getElementById('pag');
        canvas.width = 500;
        canvas.height = 500;

        let pagView = await PAG.PAGView.init(pagFile1, canvas);
        pagView.addListener('onAnimationEnd', async () => {
            start(pagView);
        });
        async function start() {
          v = v === 1 ? 2 : 1;
          pagView.setComposition(v === 2 ? pagFile2 : pagFile1)
          pagView?.setRepeatCount(1);
          pagView?.play();
        }
        start();
      }

      async function playByLite(params) {
        const { PAGView, types } = window.libpag;

        const buffer1 = await fetch('../assets/jump.pag').then((response) => response.arrayBuffer());

        const buffer2 = await fetch('../assets/hello.pag').then((response) => response.arrayBuffer());

        const canvas = document.getElementById('pag');
        canvas.width = 300;
        canvas.height = 300;

        async function start(prePagView) {
          v = v === 1 ? 2 : 1;
          let pagView = await PAGView.init(v === 2 ? buffer2 : buffer1, canvas, {
            renderingMode: types.RenderingMode.Canvas2D,
            scaleMode: types.ScaleMode.None
          });
          prePagView?.destroy?.();
          pagView?.setRepeatCount(1);
          pagView.addListener('onAnimationEnd', async () => {
            console.log('end');
            start(pagView);
          });
          pagView?.play();
        }
        start();
      }

      window.onload = async () => {
        play()
        // playByLite()
      };

    </script>
  </body>
</html>

**Reply by ligeisme** [reply]
用完整包切换素材很丝滑，但是用轻量包，两个素材的尺寸和位置都对不上了无法衔接动画

**Reply by ligeisme** [reply]
完整包效果

https://github.com/Tencent/libpag/assets/40137514/d9d1f60d-c452-4175-8d22-007dab8cf750
轻量包效果

https://github.com/Tencent/libpag/assets/40137514/fbf3cfe7-a421-45c7-a711-4a9eaf4f8504

**Comment by CodeJhF** [comment]
感谢 @zhalice2011 的回复，也抱歉我们没有及时跟进这条。

如果问题仍存在，请在最新版本（4.4 / 4.5）上复现并补充版本号与 demo，我们会优先跟进，本条先关闭。（stale-close）
